jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebAppClassLoader.java

```diff
@@ -185,15 +185,20 @@ public WebAppClassLoader(ClassLoader parent, Context context)
         String extensions = System.getProperty(WebAppClassLoader.class.getName() + ".extensions");
         if (extensions != null)
         {
-            StringTokenizer tokenizer = new StringTokenizer(extensions, ",;");
+            StringTokenizer tokenizer = new StringTokenizer(extensions, StringUtil.DEFAULT_DELIMS);
             while (tokenizer.hasMoreTokens())
             {
                 _extensions.add(tokenizer.nextToken().trim());
             }
         }
 
         if (context.getExtraClasspath() != null)
```

Comment:
I don't like WebAppClassLoader setting itself up from the extraClasspath:  the WebAppClassLoader is configured by the WebInfConfiguration, so I don't see why the extraClassPath shouldn't be too.

Reply:
We have 2 use cases ...

1. A user of WebAppContext without Configuration.  Yes, this is a thing!  sometimes accidental, most of the time intentional (see spring web, spark, akka, kotlin, etc)
2. A user of WebAppContext with Configuration.

In use case 1 we need WebAppClassLoader to configure itself.
It's super odd to have the Class you told "here, use this config" not actually be the one that uses that config.

Since this is a Jetty 10 change, we should probably step back and re-evaluate the whole WebAppContext.extraClasspath role then, as per your comments, it's really a WebInfConfiguration behavior, not a WebAppClassLoader behavior.

Reply:
In the interests of getting the rest of this applied to jetty-10, I'll drop my objection.